### ISSUE TITLE
Adding etcd

### ISSUE THREAD

**Issue by savorywatt** [tylertreat/Flotilla#2]
What do you think about using etc.d to share configuration for the messaging clusters that are spun up, so each daemon wrapper can pull down the configuration details it needs to know about how to set up the given messaging framework? 

What were your thoughts on how you wanted this implemented?

**Comment by tylertreat** [comment]
Yep, that's pretty much exactly what I was thinking. Also allowing daemons
to register themselves so the client doesn't necessarily have to specify
peers. For example, maybe I start a bunch of daemons across different
machines. It would be nice if the client didn't have to know each IP and
instead just says "run producers/consumers on 5 peers."

**Comment by savorywatt** [comment]
That would be nice, or to have a cluster of daemons register themselves with a cluster of clients all using different connection profiles (Comcast). So either way it would be nice if they could all know about each other easily.

**Comment by savorywatt** [comment]
alright, after looking this over here is my proposal

Create a new package, coordinator (lightweight service discovery server)

So to get things running  you would start a coordinator

flotilla-coordinator

Start some clients

flotilla-client --coordinator http://localhost:81818181/ --flota "testrun5"
flotilla-client --coordinator http://test.mycoolhost.com/ --flota "testrun5"
flotilla-client --coordinator 10.1.1.5 --flota "testrun5"

Start some servers

flotilla-server --coordinator 10.1.1.5 --flota "testrun5"

This is an interesting setup because it should let you be able to do a fleet based deployment of docker containers to a cluster on EC2 or wherever (using coreos of course)

https://github.com/coreos/fleet

**Comment by tylertreat** [comment]
Are you seeing the coordinator as something that would go in place of etcd? It'd be easy enough to roll something simple since it basically just needs to listen for a heartbeat from the daemons and add/remove them from the registry when they spin up or die. Just making sure I understand.

I'm trying to understand your example use case, but I'm not sure if I completely follow. The clients are still kicking off the benchmarks, right? In this case, you'd be running three different benchmarks? Why are you starting the servers after the clients?

**Comment by savorywatt** [comment]
I understand your confusion now that I have actually walked through your code in more depth. Forget the coordinator for the moment (I was thinking of embedding etcd but that is a hassle). This would be much simpler.

start etcd somewhere both the client and the server are going to be able to talk to it.

```sh
./etcd

flotilla-server --coordinator 10.1.1.5 --flota "testrun5"
```

server registers itself and establishes heartbeat. (heartbeat not really 'needed' for first pass)

could do that here
https://github.com/tylertreat/Flotilla/blob/fa5c03018a49451dd406127da7c8eebd4744b8de/flotilla-server/main.go#L31

```sh
flotilla-client --coordinator 10.1.1.5 --flota "testrun5"
```
client pulls down all active registered 'peers'

could do that here in the client
https://github.com/tylertreat/Flotilla/blob/master/flotilla-client/main.go#L55

It would be easy to expand it out eventually do share configuration details but since you are already sending out the specifics of the test from the client to the servers directly there is no reason to change all that just to have them pull it down from etcd.

**Comment by tylertreat** [comment]
Yeah, that sounds good to me.

**Comment by savorywatt** [comment]
K, I'l take a stab at it then

**Comment by savorywatt** [comment]
Initial pass here
https://github.com/savorywatt/Flotilla/tree/etcd

Ran into some flotilla-server compilation errors 

```
github.com/tylertreat/Flotilla/flotilla-server/daemon/broker/pubsub
daemon/broker/pubsub/orchestrator.go:88: cannot use oauth2.NoContext (type oauth2.Context) as type []byte in argument to google.JWTConfigFromJSON: need type assertion
daemon/broker/pubsub/orchestrator.go:88: cannot use key (type []byte) as type string in argument to google.JWTConfigFromJSON
```

When I have everything working I'll do a PR

**Comment by tylertreat** [comment]
Google loves to break their oauth2 lib...

**Comment by savorywatt** [comment]
Yeah, it was just too late to deal with it, I'll hack on it later.

**Comment by savorywatt** [comment]
Got the client pulling down the IP & port the daemon's are setting, all via etcd. Got some cleanup to do and more testing then I'll do a PR.

This fixed the google oauth2 problem https://github.com/savorywatt/Flotilla/commit/7be18010737a56c5bf95adf3f7b5d0c8390f20fe
